subsys/bluetooth/controller/ll_sw/ull_conn_iso.c

```diff
@@ -276,19 +279,55 @@ void ull_conn_iso_done(struct node_rx_event_done *done)
 
 	ticks_drift_plus  = 0;
 	ticks_drift_minus = 0;
+	handle_iter = UINT16_MAX;
+	cis = NULL;
+
+	/* Check all CISes for supervison/establishment timeout */
+	for (cis_idx = 0; cis_idx < cig->lll.num_cis; cis_idx++) {
+		cis = ll_conn_iso_stream_get_by_group(cig, &handle_iter);
+		LL_ASSERT(cis);
 
-	if (done->extra.trx_cnt) {
-		if (IS_ENABLED(CONFIG_BT_CTLR_PERIPHERAL_ISO) && lll->role) {
-			ull_drift_ticks_get(done, &ticks_drift_plus,
-					    &ticks_drift_minus);
+		if (cis->lll.handle != LLL_HANDLE_INVALID) {
+			/* CIS was setup and is now expected to be going */
+			if (!(done->extra.trx_performed_mask &
+			      (1U << LL_CIS_IDX_FROM_HANDLE(cis->lll.handle)))) {
+				/* We did NOT have successful transaction on established CIS,
+				 * or CIS was not yet established, so handle timeout
+				 */
+				if (!cis->event_expire) {
+					struct ll_conn *conn = ll_conn_get(cis->lll.acl_handle);
+
+					cis->event_expire =
+						RADIO_CONN_EVENTS(
+							conn->timeout * 10U * 1000U,
+							cig->iso_interval * CONN_INT_UNIT_US) + 1;
+				}
+
+				if (--cis->event_expire == 0) {
+					/* Stop CIS and defer cleanup to after teardown. */
+					ull_conn_iso_cis_stop(cis, NULL,
+							      cis->established ?
+							      BT_HCI_ERR_CONN_TIMEOUT :
+							      BT_HCI_ERR_CONN_FAIL_TO_ESTAB);
+
+				}
+			} else {
+				cis->event_expire = 0;
+			}
 		}
 	}
 
+	if (done->extra.trx_performed_mask &&
+	    IS_ENABLED(CONFIG_BT_CTLR_PERIPHERAL_ISO) && lll->role) {
+		ull_drift_ticks_get(done, &ticks_drift_plus,
+				    &ticks_drift_minus);
+	}
+
 	/* Update CIG ticker to compensate for drift */
-	if (ticks_drift_plus || ticks_drift_minus) {
+	if (cis && (ticks_drift_plus || ticks_drift_minus)) {
```

Comment:
I think it would be a good idea to add to the comment, that any CIS found in the above for-loop will do, as they are in the same group, and for peripheral all CISes in a CIG have the same ACL. Otherwise it may look like `cis` has a "random" value, and get's a "random" ACL.

Reply:
Ack